code/game/objects/structures/doors.dm

```diff
@@ -0,0 +1,902 @@
+///Can attach a lock
+#define DOOR_LOCKABLE 1<<0
+///Can be opened with violence
+#define DOOR_FORCEABLE 1<<1
+///If the door is open
+#define DOOR_OPEN 1<<2
+///Is currently opening
+#define DOOR_OPENING 1<<3
+///Is currently closing
+#define DOOR_CLOSING 1<<4
+
+///Set of flags your average door will most likely have
+#define STANDARD_DOOR_FLAGS DOOR_LOCKABLE|DOOR_FORCEABLE
+
+/obj/structure/door
+	name = "door"
+	desc = "It opens and closes - nothing out of the ordinary."
+	icon = 'icons/obj/doors/mineral_doors.dmi'
+	icon_state = ""
+	opacity = TRUE
+	density = TRUE
+	allow_pass_flags = PASS_AIR
+	resistance_flags = XENO_DAMAGEABLE
+	layer = DOOR_CLOSED_LAYER
+	max_integrity = 100
+	explosion_block = 0.5
+	///The name of the door, assigned at initialization; use this instead of initial(name)
+	var/base_name
```

Comment:
Want to do a labeler-related PR after this. So if you do something like rename a door to "big red door" it will transfer when it becomes almost dismantled to "big red door frame" rather than "door frame".